core/src/main/resources/lib/audio.fjs

```diff
@@ -92,48 +52,120 @@ export default (componentType: ComponentType) => ({ ophan, dom, viewport }: Serv
   };
   
   const stop = () => {
-    playPauseC.close();
-    scrubC.close();
+    playerTimeUpdateC.close();
     viewport.unobserve(root, 1, observer);
   };
 
-  const onPlayPause = (p: Audio) => (a: Action): void => {
+  const recordOphanAudioEvent = (id: string, eventName: string) => {
+    ophan.record({
+      audio: {
+        id,
+        eventType: `audio:content:${eventName}`,
+      },
+    });
+  };
+
+  const setPlayingState = (playPauseButton: HTMLElement) => {
+    dom.write(() => {
+      playPauseButton.classList.add('is-playing');
+    });
+  };
+
+  const setPausedState = (playPauseButton: HTMLElement) => {
+    dom.write(() => {
+      playPauseButton.classList.remove('is-playing');
+    });
+  };
+
+  const setupTime = (audio: Audio) => {
+    dom.write(() => {
+      audio.scrubber.value = 0;
+      audio.timePlayed.innerText = formatTime(0);
+      audio.timeDuration.innerText = formatTime(Number(audio.player.getAttribute('data-duration')) || 0);
+      audio.player.setAttribute('milestones', 'READY,');
+    });
+  };
+
+  const showProgress = (audio: Audio, percentPlayed: Number, now: Number, played: string) => {
+    const gradientDescription = `linear-gradient(to right, #C70000 ${percentPlayed}%, #afafaf ${percentPlayed}%)`;
+
+    dom.write(() => {
+      audio.scrubber.value = percentPlayed;
+      audio.player.setAttribute('last-tick', now.toString());
+      audio.scrubber.style.background = gradientDescription;
+      audio.timePlayed.innerText = formatTime(played);
+    });
+  };
+
+  const updateTime = (p: Audio) => (): void => {
+    const now = Date.now();
+    const last = p.player.getAttribute('last-tick');
+
+    // throttle to 1 update per second
+    if (now - (Number(last) || 0) < 1000) return;
+
+    const played = Math.floor(p.player.currentTime);
+    const duration = Number(p.player.getAttribute('data-duration')) || 1; // avoid divide by zero if not present
+    const percentPlayed = Math.floor(played / duration * 100);
+    const milestones = p.player.getAttribute('milestones');
```

Comment:
Just to be on the safe-side, can you please use `data-milestones` and `data-last-tick` instead?

Reply:
That should be done now. I did miss one though, and just committed it on its own.